### DISCUSSION TITLE
Extension of Auto group feature with custom user-defined groups defined via short codes after file name?

### DISCUSSION THREAD

**Discussion by michaelbeijer** [xiaoyifang/goldendict-ng#472] [Ideas]
It would be very cool if you could extend the very useful **Auto groups** feature, so that things other than language codes (nl-en, nld-eng, etc.) could be used to automatically create groups. For example, I am currently trying to devise a system to organise all my glossaries inside GoldenDict, where I am using language codes and language pairs, such as nl-en, en-nl, nl-nl, etc. However, I would also like to use a few other groups based on stuff like whether the glossary is an abbreviations resource (‘abbrev’), my own work (‘beijer’), from a specific publisher (‘oxford’), etc. 

It would be amazing if the user could add short, custom codes behind the file name that GoldenDict could use to auto-group them.

For example, I could imagine a file naming scheme such as:

Acronyms_used_in_the_CISA_Exam **(en-en, abbrev)**.dsl
Acronyms_used_in_the_CISA_Exam **[en-en, abbrev]**.dsl
Acronyms_used_in_the_CISA_Exam **{en-en, abbrev}**.dsl

… where the various automatic groups were contained in special brackets after the file name, and separated by commas.

**Comment by xiaoyifang** [comment]
good idea ,
Though I'd like to keep the filename unchanged.
Need some design about how to hold this category/structure information ,maybe an extra file called metadata.txt in the same folder as dictionary ?

**Reply by michaelbeijer** [reply]
I like the idea of a metadata.txt file!

**Comment by xiaoyifang** [comment]
https://github.com/xiaoyifang/goldendict-ng/issues/612

**Reply by michaelbeijer** [reply]
Wow, thanks a lot guys! Am currently testing it and it seems to work perfectly.
